enterprise/cmd/frontend/internal/registry/extensions_db.go

```diff
@@ -180,6 +180,48 @@ func (s dbExtensions) GetByExtensionID(ctx context.Context, extensionID string)
 	return results[0], nil
 }
 
+// Temporary: we manually set these. Featured extensions live on sourcegraph.com, all other instances ask
+// dotcom for these extensions and filter based on site configuration.
+var featuredExtensionIDs = []string{"sourcegraph/codecov", "sourcegraph/sentry", "sourcegraph/vscode-extras"}
+
+// GetFeaturedExtensions retrieves the set of currently featured extensions.
+// This should only be called on dotcom; all other instances should retrieve these
+// extensions from dotcom through the HTTP API.
+//
+// 🚨 SECURITY: The caller must ensure that the actor is permitted to view these registry extensions.
+func (s dbExtensions) GetFeaturedExtensions(ctx context.Context) ([]*dbExtension, error) {
+	return s.getFeaturedExtensions(ctx, featuredExtensionIDs)
+}
+
+func (s dbExtensions) getFeaturedExtensions(ctx context.Context, featuredExtensionIDs []string) ([]*dbExtension, error) {
+	if mocks.extensions.GetFeaturedExtensions != nil {
+		return mocks.extensions.GetFeaturedExtensions()
+	}
+
+	conds := []*sqlf.Query{}
```

Comment:
```suggestion
	conds := make([]*sqlf.Query, 0, len(featuredExtensionIDs))
```

This would avoid many reallocations in case of a large slice, because we know at this point, we want to append _at most_ `len(featuredExtensionIDs)` elements.